### PR TITLE
fix: properly load Qt and Webengine's built in translations, delete GD's custom ones

### PR DESCRIPTION
**Pull request by shenlebantongying**
The `qt_extra_` were added 10 years ago to do something that's out of the scope of GD project.

They aren't updated at all, and they should have been added https://github.com/qt/qttranslations instead.

Quick check shows that at least `it` language is already translated on Qt's side.

https://l10n-files.qt.io/l10n-files/ .

---

The `qtwebengine_zh_CN` is already translated by you https://github.com/qt/qttranslations/blob/dev/translations/qtwebengine_zh_CN.ts 2 years ago

My quick decompiling Windows installer's `qt_zh_CN.qm` file shows that the strings from `qtwebengine_zh_CN` are included.

The `windeployqt` will combine various qt modules into one, hence just load `qt_*` is enough.

Untested on Windows, but on macOS, this is the result with `en_CA` local.
```
qt translator loaded:  "/opt/homebrew/share/qt/translations/qt_en.qm"
qt webengine translator loaded:  "/opt/homebrew/share/qt/translations/qtwebengine_en.qm"
```

---

Also handles `[nodiscard]` compile warnings religiously.

### CODE REVIEW

File: src/main.cc [hunk 1]
```diff
@@ -564,19 +564,36 @@ int main( int argc, char ** argv )
   QLocale::setDefault( locale );
   QApplication::setLayoutDirection( locale.textDirection() );
 
-  if ( !qtTranslator.load( "qt_extra_" + localeName, Config::getLocDir() ) ) {
-    qtTranslator.load( "qt_extra_" + localeName, QLibraryInfo::location( QLibraryInfo::TranslationsPath ) );
+  // Load Qt translators
+  // For Windows, windeployqt will combine multiple qt modules translations into `qt_*`
+  // Thus, after deployment, loading `qtwebengine_*` is guaranteed to fail on Windows.
```
Comment (xiaoyifang) [inline src/main.cc:569]: the loading qtwebengine_* should be used to load the `local/qtwebengine_zh_CN.ts` file? I'm not sure right now.

Comment (shenlebantongying) [reply src/main.cc:569]: No, the one you translated is already included in standard Qt installations.

**Comment by xiaoyifang** [comment]
Can be tested on Windows ,and  check the right-mouse context menu on the page? Does it still have untranslated items there?

**Comment by shenlebantongying** [comment]
I cannot test windows now, but checking what's inside `qt_zh_CN.qm` in the installers shows that they are there.

**Comment by shenlebantongying** [comment]
https://github.com/qt/qttranslations has `es` `it` `lt`. It is unlikely the original GD's version from 10 years ago is better. Even if it is, they should be upstreamed instead of staying in this repo.

**Comment by shenlebantongying** [comment]
I tested with macOS version. The macdeployqt also combines all translations into `qt_` and `zh_CH` webengine translations are there.

https://github.com/xiaoyifang/goldendict-ng/pull/1918

**Pull request merged**